src/nr_openai_observability/langchain_callback.py

```diff
@@ -207,6 +208,11 @@ def _start_segment(self, run_id, trace, tags={}):
             trace.add_custom_attribute(key, val)
 
         stack = self.trace_stacks.get(run_id, [])
+
+        trace.parent = self.parent_trace
```

Comment:
Does this cause a problem if the callback handler is initialized once and shared? Also, do we want to mess with the `trace`'s parents? I was assuming the issue was in the the spans / segments getting incorrect parents?